### PR TITLE
WIP Reconfig to snake 2

### PR DESCRIPTION
**Pull request by Trphajzlik**

### CODE REVIEW

**Comment by Trphajzlik** [comment]
Implemented connecting one arm with another, given there is free space between arms. Need to check how well it will work, for now I've checked it with this config:
`C
M 0 90 0 0
M 1 0 90 0
M 2 0 0 90
M 3 0 0 0
M 4 0 0 0
M 5 90 0 0
M 6 0 0 90
M 7 0 0 0
M 8 0 0 0
M 9 0 0 0
E 0 0 2 0 2 1 1
E 1 0 2 0 2 1 2
E 2 0 1 0 0 1 3
E 3 0 2 0 2 1 4
E 0 1 2 0 2 0 5
E 5 1 2 0 2 0 6
E 6 1 0 0 1 0 7
E 5 0 0 0 2 0 8
E 5 1 0 0 2 0 9`

- [ ] Need to check `SnakeStar`, I've changed something that looked like bug

**Comment by Trphajzlik** [comment]
makeSpace for now takes too long to finish (~40 minutes for 25-module robot)

- [ ] redo makeSpace -- maybe changing params will be enough (smaller queue, maybe if the limit would be just equal to `number_of_module`, it would still produce "good enough" results. Second option is just to make space around the `isolate` modules -- problem with this is that it doesnt force other branches to make space for the branches right around `isolate` modules. (This func is needed as prep function for `fixLeaf`.)

- [ ] add second version of makeSpace --give it two branches and maybe try to minimalize number of filled spaces between those branches. (This func is needed as prep function for `connectArms`)

File: RoFILib/snake_reconfig/main.cpp [hunk 1]
```diff
@@ -6,38 +6,90 @@
...
+    }
+}
+
 int main(int argc, char* argv[])
 {
     if (argc < 2) {
         throw std::invalid_argument("No filepath given");
     }
-
+/*
```
Comment (yaqwsx) [inline RoFILib/snake_reconfig/main.cpp:47]: Please remove all the commented code

**Review by yaqwsx — CHANGES_REQUESTED** [review]
I briefly walked through the code. It is ok for me to merge it. 

It makes no sense to preserve most of the commits (since they contain misleading naming or they are just fixes). I suggest to squash them even in a single commit or separate the code and benchmarks with input configuration into separate commits.

**Pull request merged**